src/python/pants/backend/python/util_rules/pex_test.py

```diff
@@ -535,3 +536,67 @@ def assert_description(
         use_repo_pex=True,
         expected="Extracting all requirements in lock.txt from repo.pex to build new.pex",
     )
+
+
+def test_error_on_invalid_lockfile_with_path(rule_runner: RuleRunner) -> None:
+    with pytest.raises(ExecutionError):
+        _run_pex_for_lockfile_test(
+            rule_runner, True, actual="1bad", expected="900d", behavior="error"
+        )
+
+
+def test_warn_on_invalid_lockfile_with_path(rule_runner: RuleRunner, caplog) -> None:
+    _run_pex_for_lockfile_test(rule_runner, True, actual="1bad", expected="900d", behavior="warn")
+    assert "Invalid lockfile provided." in caplog.text
+
+
+def test_no_warning_on_valid_lockfile_with_path(rule_runner: RuleRunner, caplog) -> None:
+    _run_pex_for_lockfile_test(rule_runner, True, actual="900d", expected="900d", behavior="warn")
+    assert not caplog.text.strip()
+
+
+def test_error_on_invalid_lockfile_with_content(rule_runner: RuleRunner) -> None:
+    with pytest.raises(ExecutionError):
+        _run_pex_for_lockfile_test(
+            rule_runner, False, actual="1bad", expected="900d", behavior="error"
+        )
+
+
+def test_warn_on_invalid_lockfile_with_content(rule_runner: RuleRunner, caplog) -> None:
+    _run_pex_for_lockfile_test(rule_runner, False, actual="1bad", expected="900d", behavior="warn")
+    assert "Invalid lockfile provided." in caplog.text
+
+
+def test_no_warning_on_valid_lockfile_with_content(rule_runner: RuleRunner, caplog) -> None:
+    _run_pex_for_lockfile_test(rule_runner, False, actual="900d", expected="900d", behavior="warn")
+    assert not caplog.text.strip()
+
+
+def _run_pex_for_lockfile_test(rule_runner, use_file, actual, expected, behavior):
+    lockfile = f"""
+# --- BEGIN PANTS LOCKFILE METADATA: DO NOT EDIT OR REMOVE ---
+# invalidation digest: {actual}
+# --- END PANTS LOCKFILE METADATA ---
+ansicolors==1.1.8
+"""
+    if use_file:
+        rule_runner.create_file("lockfile.txt", lockfile)
```

Comment:
I've been trying to actively kill `.create_file()` - almost there.

```suggestion
        rule_runner.write_files({"lockfile.txt": lockfile})
```

Reply:
👍